File: src/main/java/de/fluxparticle/animation/Timeframe.java
package de.fluxparticle.animation;

import de.fluxparticle.animation.signal.SignalEvent;
import javafx.util.Duration;

import java.util.Arrays;
import java.util.Collection;

public class Timeframe {

	private final Duration duration;

	private final Collection<SignalEvent<?>> signalEvents;

	public Timeframe(Duration duration, SignalEvent<?>... signalEvents) {
		this.duration = duration;
		this.signalEvents = Arrays.asList(signalEvents);
	}

	public Duration getDuration() {
		return duration;
	}

	public Collection<SignalEvent<?>> getSignalEvents() {
		return signalEvents;
	}

}

File: src/main/kotlin/de/fluxparticle/animation/Timeframe.kt
package de.fluxparticle.animation

import de.fluxparticle.animation.signal.SignalEvent

class Timeframe(val duration: Double, vararg signalEvents: SignalEvent<out Any>) {

    val signalEvents: Collection<SignalEvent<out Any>>

    init {
        this.signalEvents = listOf(*signalEvents)
    }

}
